error.ts

```diff
@@ -44,51 +50,57 @@ export class KeyboardInterrupt extends Error {
 
 export class Exception extends Error {
   __proto__: Error;
-  constructor(message?: string, name = "Exception") {
+  loc: Location;
+
+  constructor(message?: string, name = "Exception", loc?: Location) {
     const trueProto = new.target.prototype;
     super(message);
     this.name = name;
+    this.loc = loc;
 
     // Maintains proper stack trace for where our error was thrown (only available on V8)
     if (Error.captureStackTrace) {
       Error.captureStackTrace(this, Exception);
     }
-
     // Alternatively use Object.setPrototypeOf if you have an ES6 environment.
     this.__proto__ = trueProto;
   }
 }
 
 export class StopIteration extends Exception {
-  constructor(message?: string, name = "StopIteration") {
-    super(message, name);
+  constructor(message?: string, loc?: Location) {
```

Comment:
Do you anticipate having lots of cases where `loc` is in fact not provided to the constructors?

Reply:
Because at the beginning we assumed every group would merge their implementation during this milestone. So we planned to make it optional and then help other groups add location in parser.ts after complete merge.
We are making it required and also move it into the "a" tag.

Reply:
Got it, good idea.